src/App.js

```diff
@@ -14,11 +18,140 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.PureComponent {
+  state = {
+    goods: [...goodsFromServer],
+    loading: false,
+    sortAlphabet: false,
+    sortLength: true,
+    select: 1,
+    selectedValues: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+  }
+
+  startLoading = () => {
+    this.setState({
+      loading: true,
+    });
+  }
+
+  handleReverse = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  }
+
+  handleAlphabetSort = () => {
+    const { sortAlphabet } = this.state;
+
+    sortAlphabet
+      ? (
+        this.setState(state => ({
+          goods: [...state.goods].sort(
+            (a, b) => a.localeCompare(b),
+          ),
+          sortAlphabet: false,
+        }))
+      )
+      : (
+        this.setState(state => ({
+          goods: [...state.goods].sort(
+            (a, b) => b.localeCompare(a),
+          ),
+          sortAlphabet: true,
+        }))
+      );
```

Comment:
```suggestion
    this.setState(state => ({
          goods: [...state.goods].sort(
            (a, b) => {
              return sortAlphabet
                ? a.localeCompare(b)
                : b.localeCompare(a);
            },
          ),
          sortAlphabet: false,
        }))
```
щось типу цього має спрацювати